pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -165,44 +172,93 @@ def get_queryset(self):
         """
         Returns a Collections queryset for specified distribution.
         """
-        repo_version = self.get_repository_version(self.kwargs["path"])
-        deprecated_query = AnsibleCollectionDeprecated.objects.filter(
-            collection=OuterRef("pk"), repository_version=repo_version
+        repo_version = self._repository_version
+        return Collection.objects.filter(
+            versions__in=repo_version.content,
         )
-        collections = Collection.objects.filter(versions__in=repo_version.content).distinct()
-        collections = collections.annotate(deprecated=Exists(deprecated_query))
-
-        versions_qs = CollectionVersion.objects.filter(pk__in=repo_version.content).values_list(
-            "collection_id",
-            "namespace",
-            "name",
-            "version",
-            "pulp_created",
+
+    def append_context(self, queryset):
```

Comment:
This code is sufficiently complicated that I believe we have to look at other options to accomplish this within the runtime goals of our users, e.g. galaxy_ng. To do that I believe we'll have to denormalize the db and not try to perform this computation at runtime anymore.

So in terms of moving forward I see two options:
a) merge this and de-normalize later
b) don't merge this and de-normalize later

I lean towards option (a) because it does make an improvement even though it's basically unmaintainable.

@fao89 wdyt?

Reply:
I agree!
Let's put the de-nomarlization on the agenda for the next meeting

Reply:
OK then let's merge this for now and we'll try to address this in the future.